.github/workflows/rust-appencryption-bench.yml

```diff
@@ -0,0 +1,42 @@
+name: '[Rust] AppEncryption Benchmarks'
+
+on:
+  workflow_dispatch:
+    inputs:
+      rust-version:
+        description: 'Rust toolchain version'
+        required: false
+        default: 'stable'
+
+  schedule:
+    - cron: '0 0 * * 0'  # Run weekly on Sunday at midnight
+
+permissions:
+  contents: read
+
+jobs:
+  benchmark:
+    name: Run Benchmarks
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
```

Comment:
Afaik ubuntu latest already comes with Rust, allowing you to run `rustup component add rustfmt clippy` and similar. To then directly use `cargo` to run tests, lints or benchmarks